### PR TITLE
test: Migrate LSP7DigitalAsset & LSP8Identifiable DigitalAsset & LSP9Vault tests to mocha

### PR DESCRIPTION
**Pull request by b00ste**
## What does this PR introduce?
Migrating the tests for the LSP7DigitalAsset, LSP8IdentifiableDigitalAsset and LSP9Vault from jest to mocha, using expect from chai.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,11 +21,11 @@
     "test:up": "NODE_NO_WARNINGS=1 hardhat test --no-compile tests/UniversalProfile.test.ts",
     "test:lsp1": "NODE_NO_WARNINGS=1 hardhat test --no-compile tests/LSP1UniversalReceiver/*.test.ts",
     "test:lsp2": "NODE_NO_WARNINGS=1 hardhat test --no-compile tests/LSP2ERC725YJSONSchema/LSP2UtilsLibrary.test.ts",
-    "test:lsp4": "NODE_NO_WARNINGS=1 hardhat test --no-compile tests/LSP4DigitalAssetMetadata/LSP4Compatibility.test.ts",
+    "test:lsp4": "NODE_NO_WARNINGS=1  tests/LSP4DigitalAssetMetadata/LSP4Compatibility.test.ts",
```
Comment (YamenMerhi) [inline package.json:24]: you cut `hardhat test --no-compile`, please re-add

**Review by YamenMerhi — APPROVED** [review]

**Review by YamenMerhi — APPROVED** [review]

File: tests/LSP7DigitalAsset/extensions/LSP7CompatibleERC20.behaviour.ts [hunk 2]
```diff
@@ -519,44 +543,41 @@ export const shouldInitializeLikeLSP7CompatibleERC20 = (
...
+      ).withArgs(SupportedStandards.LSP4DigitalAsset.key);
       expect(
         await context.lsp7CompatibleERC20["getData(bytes32)"](
           SupportedStandards.LSP4DigitalAsset.key
         )
-      ).toEqual(SupportedStandards.LSP4DigitalAsset.value);
+      ).to.be.equal(SupportedStandards.LSP4DigitalAsset.value);
 
       const nameKey =
         "0xdeba1e292f8ba88238e10ab3c7f88bd4be4fac56cad5194b6ecceaf653468af1";
```
Comment (CJ42) [inline tests/LSP7DigitalAsset/extensions/LSP7CompatibleERC20.behaviour.ts:557]: I suggest that we use the values from `constants.js` and avoid hex literals in tests for data keys, as these are important (also to just defined them in one place, if they ever need to be changed).

add at the top:

```ts
import { ERC725YKeys } from "../../constants.js"
```

and change by

```suggestion
       ERC725YKeys.LSP4["LSP4TokenName"]
```

or 

```suggestion
        ERC725YKeys.LSP4.LSP4TokenName
```

> I prefer the 1st one to differentiate the LSP standard and the data key, but it is up to you, both syntax are valid :)

File: tests/LSP7DigitalAsset/extensions/LSP7CompatibleERC20.behaviour.ts [hunk 3]
```diff
@@ -519,44 +543,41 @@ export const shouldInitializeLikeLSP7CompatibleERC20 = (
...
-      );
+        "DataChanged"
+        ).withArgs(nameKey);
       expect(
         await context.lsp7CompatibleERC20["getData(bytes32)"](nameKey)
-      ).toEqual(expectedNameValue);
+      ).to.be.equal(expectedNameValue);
 
       const symbolKey =
         "0x2f0a68ab07768e01943a599e73362a0e17a63a72e94dd2e384d2c1d4db932756";
```
Comment (CJ42) [inline tests/LSP7DigitalAsset/extensions/LSP7CompatibleERC20.behaviour.ts:570]: Same comment as above

```suggestion
        ERC725YKeys.LSP4["LSP4TokenSymbol"];
```

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { ethers } from "hardhat";
```
Comment (CJ42) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts:2]: @YamenMerhi @b00ste 

A small detail, but this file is named `.behavior.ts` (US syntax) while the other are `*.behaviour.ts` (UK syntax).

How about sticking to `behaviour.ts`? Let's agree on the common syntax to be consistent 🙂

File: tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts [hunk 5]
```diff
@@ -57,7 +58,7 @@ export const shouldBehaveLikeLSP7Mintable = (
         context.accounts.tokenReceiver.address
       );
 
-      expect(tokenReceiverBalance.toNumber()).toEqual(amountToMint.toNumber());
+      expect(tokenReceiverBalance.toNumber()).to.be.equal(amountToMint.toNumber());
```
Comment (CJ42) [inline tests/LSP7DigitalAsset/extensions/LSP7Mintable.behavior.ts:61]: The chai matchers from Hardhat now work with BN, so no need anymore to convert `.toNumber()`

```suggestion
      expect(tokenReceiverBalance).to.be.equal(amountToMint);
```

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8CappedSupply.behaviour.ts [hunk 6]
```diff
@@ -43,15 +44,15 @@ export const shouldBehaveLikeLSP8CappedSupply = (
...
+      expect(tokenSupplyCap).to.be.equal(context.deployParams.tokenSupplyCap);
     });
   });
 
   describe("when minting tokens", () => {
     it("should allow minting amount up to tokenSupplyCap", async () => {
       const preTokenSupplyCap = await context.lsp8CappedSupply.tokenSupplyCap();
       const preTotalSupply = await context.lsp8CappedSupply.totalSupply();
-      expect(preTokenSupplyCap.sub(preTotalSupply).toString()).toEqual(
+      expect(preTokenSupplyCap.sub(preTotalSupply).toString()).to.be.equal(
```
Comment (CJ42) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8CappedSupply.behaviour.ts:55]:
```suggestion
      expect(preTokenSupplyCap.sub(preTotalSupply)).to.be.equal(
```

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8CappedSupply.behaviour.ts [hunk 7]
```diff
@@ -103,7 +104,7 @@ export const shouldBehaveLikeLSP8CappedSupply = (
             context.accounts.tokenReceiver.address,
             anotherTokenId
           )
-        ).toBeRevertedWith("LSP8CappedSupplyCannotMintOverCap()");
+        ).to.be.revertedWith("LSP8CappedSupplyCannotMintOverCap()");
```
Comment (CJ42) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8CappedSupply.behaviour.ts:107]:
```suggestion
        ).to.be.revertedWithCustomError("LSP8CappedSupplyCannotMintOverCap");
```

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Enumerable.behavior.ts [hunk 8]
```diff
@@ -1,5 +1,7 @@
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
```
Comment (CJ42) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Enumerable.behavior.ts:1]: Same comment as above, let's rename to `behaviour.ts`

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts [hunk 9]
```diff
@@ -1,5 +1,6 @@
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
```
Comment (CJ42) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts:1]: Same as above, let's rename to `behaviour.ts` ✅ , not ~`behavior.ts`~ ❌

File: tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts [hunk 10]
```diff
@@ -47,15 +48,15 @@ export const shouldBehaveLikeLSP8Mintable = (
...
+      expect(postMintTotalSupply).to.be.equal(preMintTotalSupply.add(1));
     });
 
     it("tokenReceiver balance should have increased", async () => {
       const tokenReceiverBalance = await context.lsp8Mintable.balanceOf(
         context.accounts.tokenReceiver.address
       );
 
-      expect(tokenReceiverBalance.toNumber()).toEqual(1);
+      expect(tokenReceiverBalance.toNumber()).to.be.equal(1);
```
Comment (CJ42) [inline tests/LSP8IdentifiableDigitalAsset/extensions/LSP8Mintable.behavior.ts:59]:
```suggestion
      expect(tokenReceiverBalance).to.be.equal(1);
```

**Review by CJ42 — CHANGES_REQUESTED** [review]
Awesome job! 🚀 

I have added some review comment.

I would suggest overall that we use `to.equal` ✅ instead of ~`to.be.equal`~ ❌ 

See the chai API reference: https://www.chaijs.com/api/bdd/#method_equal

File: tests/LSP7DigitalAsset/LSP7DigitalAsset.behaviour.ts [hunk 11]
```diff
@@ -405,40 +421,46 @@ export const shouldBehaveLikeLSP7 = (
...
 
           if (operator.address !== from) {
             const postIsOperatorFor = await context.lsp7.isOperatorFor(
               operator.address,
               from
             );
-            expect(postIsOperatorFor).toEqual(preIsOperatorFor.sub(amount));
+            expect(postIsOperatorFor).to.be.equal(preIsOperatorFor.sub(amount));
 
             if (postIsOperatorFor.eq("0")) {
```
[the remark this answers is not in the document]
Comment (b00ste) [reply tests/LSP7DigitalAsset/LSP7DigitalAsset.behaviour.ts:448]: It shows as BigNumber and this `postIsOperatorFor == false` seems to be invalid

**Review by CJ42 — APPROVED** [review]

**Pull request merged**